### ISSUE TITLE
Crash in Text

### ISSUE THREAD

**Issue by rjkroege** [rjkroege/edwood#68]
Sigh.

```
goroutine 36 [running]:
main.(*Text).Delete(0xc42021ed30, 0x2af, 0x30c, 0x1)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/text.go:558 +0x56a
main.cut(0xc42021ed30, 0xc42021ed30, 0x0, 0x101, 0x0, 0x0)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/exec.go:321 +0x31d
main.(*Text).Type(0xc42021ed30, 0xc400000020)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/text.go:886 +0x205
main.(*Window).Type(0xc42021ec00, 0xc42021ed30, 0x20)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/wind.go:417 +0x37
main.(*Row).Type(0x12f4600, 0x20, 0x1af, 0x164, 0xc42021ed30)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/row.go:285 +0x139
main.keyboardthread(0xc4200d40c0)
```

**Comment by rjkroege** [comment]
oops. Left out the actual error:

```
panic: text.delete
```

at the very top of `Text.Delete`

**Comment by rjkroege** [comment]
Happened again. This bug is seriously annoying.

```
Untested: main.run(0x0, 0xc4200169c4, 0x3, 0xc420020300, 0x3b, 0xc4201d8001, 0x0, 0x0, 0x0, 0x0, ...): /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/exec.go:675 +0x34
Untested: main.run(0x0, 0xc42045d7a4, 0x4, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, ...): /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/exec.go:675 +0x34
acme: text.type: <nil>
panic: text.delete

goroutine 8 [running]:
main.(*Text).Delete(0xc420070530, 0x1ba, 0x329, 0x101)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/text.go:556 +0x56a
main.cut(0xc420070530, 0xc420070530, 0x0, 0x101, 0x0, 0x0)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/exec.go:321 +0x31d
main.(*Text).Type(0xc420070530, 0xc40000007d)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/text.go:884 +0x205
main.(*Window).Type(0xc420070400, 0xc420070530, 0x7d)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/wind.go:417 +0x37
main.(*Row).Type(0x12f8600, 0x7d, 0x1b, 0x340, 0xc420070530)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/row.go:289 +0x139
main.keyboardthread(0xc4200da0c0)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/acme.go:467 +0x1e8
created by main.main
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/acme.go:175 +0x974
```

**Comment by rjkroege** [comment]
Some observations:

* This bug happens when typing
* The `panic` fires when ` t.ncache is != 0` and `tofile` is true

I suspect that `t.q0` and/or `t.q1` are. being modified asynchronously and we should have never gotten to this point in the code.

**Comment by rjkroege** [comment]
Closed. But not nicely. I'll fix it more nicely in #74
